From 3480190b5a656cf0311dc2bbef0b59b6d9435a12 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Fri, 26 Jul 2013 14:04:31 +0300
Subject: [PATCH 22/24] fw: Make firmware timeout loading value configurable.

The default 60 seconds is just too much. Introduce an config
option to set it.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
---
 drivers/base/Kconfig          | 7 +++++++
 drivers/base/firmware_class.c | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/base/Kconfig b/drivers/base/Kconfig
index c8b4539..02ad36f 100644
--- a/drivers/base/Kconfig
+++ b/drivers/base/Kconfig
@@ -145,6 +145,13 @@ config EXTRA_FIRMWARE_DIR
 	  this option you can point it elsewhere, such as /lib/firmware/ or
 	  some other directory containing the firmware files.
 
+config FIRMWARE_LOADING_TIMEOUT
+	int "Default Firmware loading timeout in seconds"
+	depends on FW_LOADER
+	default 60
+	help
+	  Defines the timeout value of firmware loading in seconds
+
 config DEBUG_DRIVER
 	bool "Driver Core verbose debug messages"
 	depends on DEBUG_KERNEL
diff --git a/drivers/base/firmware_class.c b/drivers/base/firmware_class.c
index b392b35..acec6db 100644
--- a/drivers/base/firmware_class.c
+++ b/drivers/base/firmware_class.c
@@ -93,7 +93,7 @@ enum fw_buf_fmt {
 	PAGE_BUF,	/* used in loading via userspace */
 };
 
-static int loading_timeout = 60;	/* In seconds */
+static int loading_timeout = CONFIG_FIRMWARE_LOADING_TIMEOUT;	/* In seconds */
 
 static inline long firmware_loading_timeout(void)
 {
-- 
1.9.0

